### ISSUE TITLE
plotFingerprint fails with too many bam files

### ISSUE THREAD

**Issue by bc2zb** [nf-core/chipseq#70]
With  experiment of 36 samples, `plotFingerprint` fails. Scaling back plotFingerprint to 3 bam files removed error. Should be more rigorously tested to figure out maximum number of bam files or if one of my bam files was corrupted. Ideally, pipeline would use `macs.config` file to `plotFingerprint` in groups or just each sample against its respective input. 

Example error section:

```
ERROR ~ Error executing process > 'deepTools (2018_051_S5_R1_001.dedup.sorted)'
Caused by:
  Process `deepTools (2018_051_S5_R1_001.dedup.sorted)` terminated with an error exit status (1)
Command executed:
  plotFingerprint \
      -b 2018_051_S5_R1_001.dedup.sorted.bam 2018_054_S7_R1_001.dedup.sorted.bam 2018_046_S1_R1_001.dedup.sorted.bam 2018_037_S1_R1_001.dedup.sorted.bam 2018_042_S9_R1_001.dedup.sorted.bam 2018_043_S6_R1_001.dedup.sorted.bam 2018_040_S4_R1_001.dedup.sorted.bam 2018_038_S2_R1_001.dedup.sorted.bam 2018_053_S8_R1_001.dedup.sorted.bam 2018_048_S9_R1_001.dedup.sorted.bam 2018_044_S7_R1_001.dedup.sorted.bam 2018_039_S3_R1_001.dedup.sorted.bam 2018_052_S6_R1_001.dedup.sorted.bam 2018_045_S8_R1_001.dedup.sorted.bam 2018_050_S4_R1_001.dedup.sorted.bam 2018_041_S5_R1_001.dedup.sorted.bam 2018_047_S2_R1_001.dedup.sorted.bam 2018_049_S3_R1_001.dedup.sorted.bam 2018_030_S3_R1_001.dedup.sorted.bam 2018_034_S7_R1_001.dedup.sorted.bam 2018_033_S6_R1_001.dedup.sorted.bam 2018_031_S4_R1_001.dedup.sorted.bam 2018_035_S8_R1_001.dedup.sorted.bam 2018_029_S2_R1_001.dedup.sorted.bam 2018_036_S9_R1_001.dedup.sorted.bam 2018_032_S5_R1_001.dedup.sorted.bam 2018_028_S1_R1_001.dedup.sorted.bam 2018_022_S4_R1_001.dedup.sorted.bam 2018_027_S9_R1_001.dedup.sorted.bam 2018_021_S3_R1_001.dedup.sorted.bam 2018_024_S6_R1_001.dedup.sorted.bam 2018_020_S2_R1_001.dedup.sorted.bam 2018_023_S5_R1_001.dedup.sorted.bam 2018_019_S1_R1_001.dedup.sorted.bam 2018_026_S8_R1_001.dedup.sorted.bam 2018_025_S7_R1_001.dedup.sorted.bam \
      --plotFile fingerprints.pdf \
      --outRawCounts fingerprint.txt \
      --extendReads 100 \
      --skipZeros \
      --ignoreDuplicates \
      --numberOfSamples 50000 \
      --binSize 500 \
      --plotFileFormat pdf \
      --plotTitle "Fingerprints"
  
  for bamfile in 2018_051_S5_R1_001.dedup.sorted.bam 2018_054_S7_R1_001.dedup.sorted.bam 2018_046_S1_R1_001.dedup.sorted.bam 2018_037_S1_R1_001.dedup.sorted.bam 2018_042_S9_R1_001.dedup.sorted.bam 2018_043_S6_R1_001.dedup.sorted.bam 2018_040_S4_R1_001.dedup.sorted.bam 2018_038_S2_R1_001.dedup.sorted.bam 2018_053_S8_R1_001.dedup.sorted.bam 2018_048_S9_R1_001.dedup.sorted.bam 2018_044_S7_R1_001.dedup.sorted.bam 2018_039_S3_R1_001.dedup.sorted.bam 2018_052_S6_R1_001.dedup.sorted.bam 2018_045_S8_R1_001.dedup.sorted.bam 2018_050_S4_R1_001.dedup.sorted.bam 2018_041_S5_R1_001.dedup.sorted.bam 2018_047_S2_R1_001.dedup.sorted.bam 2018_049_S3_R1_001.dedup.sorted.bam 2018_030_S3_R1_001.dedup.sorted.bam 2018_034_S7_R1_001.dedup.sorted.bam 2018_033_S6_R1_001.dedup.sorted.bam 2018_031_S4_R1_001.dedup.sorted.bam 2018_035_S8_R1_001.dedup.sorted.bam 2018_029_S2_R1_001.dedup.sorted.bam 2018_036_S9_R1_001.dedup.sorted.bam 2018_032_S5_R1_001.dedup.sorted.bam 2018_028_S1_R1_001.dedup.sorted.bam 2018_022_S4_R1_001.dedup.sorted.bam 2018_027_S9_R1_001.dedup.sorted.bam 2018_021_S3_R1_001.dedup.sorted.bam 2018_024_S6_R1_001.dedup.sorted.bam 2018_020_S2_R1_001.dedup.sorted.bam 2018_023_S5_R1_001.dedup.sorted.bam 2018_019_S1_R1_001.dedup.sorted.bam 2018_026_S8_R1_001.dedup.sorted.bam 2018_025_S7_R1_001.dedup.sorted.bam
  do
      bamCoverage \
        -b $bamfile \
        --extendReads 100 \
        --normalizeUsing RPKM \
        -o ${bamfile}.bw
  done
  
  multiBamSummary \
      bins \
      --binSize 10000 \
      --bamfiles 2018_051_S5_R1_001.dedup.sorted.bam 2018_054_S7_R1_001.dedup.sorted.bam 2018_046_S1_R1_001.dedup.sorted.bam 2018_037_S1_R1_001.dedup.sorted.bam 2018_042_S9_R1_001.dedup.sorted.bam 2018_043_S6_R1_001.dedup.sorted.bam 2018_040_S4_R1_001.dedup.sorted.bam 2018_038_S2_R1_001.dedup.sorted.bam 2018_053_S8_R1_001.dedup.sorted.bam 2018_048_S9_R1_001.dedup.sorted.bam 2018_044_S7_R1_001.dedup.sorted.bam 2018_039_S3_R1_001.dedup.sorted.bam 2018_052_S6_R1_001.dedup.sorted.bam 2018_045_S8_R1_001.dedup.sorted.bam 2018_050_S4_R1_001.dedup.sorted.bam 2018_041_S5_R1_001.dedup.sorted.bam 2018_047_S2_R1_001.dedup.sorted.bam 2018_049_S3_R1_001.dedup.sorted.bam 2018_030_S3_R1_001.dedup.sorted.bam 2018_034_S7_R1_001.dedup.sorted.bam 2018_033_S6_R1_001.dedup.sorted.bam 2018_031_S4_R1_001.dedup.sorted.bam 2018_035_S8_R1_001.dedup.sorted.bam 2018_029_S2_R1_001.dedup.sorted.bam 2018_036_S9_R1_001.dedup.sorted.bam 2018_032_S5_R1_001.dedup.sorted.bam 2018_028_S1_R1_001.dedup.sorted.bam 2018_022_S4_R1_001.dedup.sorted.bam 2018_027_S9_R1_001.dedup.sorted.bam 2018_021_S3_R1_001.dedup.sorted.bam 2018_024_S6_R1_001.dedup.sorted.bam 2018_020_S2_R1_001.dedup.sorted.bam 2018_023_S5_R1_001.dedup.sorted.bam 2018_019_S1_R1_001.dedup.sorted.bam 2018_026_S8_R1_001.dedup.sorted.bam 2018_025_S7_R1_001.dedup.sorted.bam \
      -out multiBamSummary.npz \
      --extendReads 100 \
      --ignoreDuplicates \
      --centerReads
  
  plotCorrelation \
      -in multiBamSummary.npz \
      -o scatterplot_PearsonCorr_multiBamSummary.png \
      --outFileCorMatrix scatterplot_PearsonCorr_multiBamSummary.txt \
      --corMethod pearson \
      --skipZeros \
      --removeOutliers \
      --plotTitle "Pearson Correlation of Read Counts" \
      --whatToPlot scatterplot
  
  plotCorrelation \
      -in multiBamSummary.npz \
      -o heatmap_SpearmanCorr_multiBamSummary.png \
      --outFileCorMatrix heatmap_SpearmanCorr_multiBamSummary.txt \
      --corMethod spearman \
      --skipZeros \
      --plotTitle "Spearman Correlation of Read Counts" \
      --whatToPlot heatmap \
      --colorMap RdYlBu \
      --plotNumbers
  
  plotPCA \
      -in multiBamSummary.npz \
      -o pcaplot_multiBamSummary.png \
      --plotTitle "Principal Component Analysis Plot" \
      --outFileNameData pcaplot_multiBamSummary.txt
Command exit status:
  1
Command output:
  (empty)
Command error:
  Traceback (most recent call last):
    File "/usr/local/apps/deeptools/3.1.3/bin/plotFingerprint", line 12, in <module>
      main(args)
    File "/usr/local/Anaconda/envs_app/deeptools/3.1.3/lib/python3.6/site-packages/deeptools/plotFingerprint.py", line 421, in main
      plt.plot(x, count, label=args.labels[i], linestyle=pyplot_line_styles[j])
  IndexError: list index out of range
```

**Comment by drpatelh** [comment]
Maybe it's worth following this up with the authors of `deepTools` to make sure that there is an upper limit in terms of the number of samples. Would be better if an appropriate error message was generated before the rest of the script is executed.

It's not that informative having this many samples plotted for this kind of assessment. Ideally, we would plot all the IPs from the same experimental group along with their corresponding inputs.

**Comment by ewels** [comment]
Looks like this has been fixed in deepTools pretty recently: https://github.com/deeptools/deepTools/commit/be9eda61d4658be872e14d2a24302a1bc9a747b1

If we update the pipeline to v3.2.0 of deepTools then the error should be resolved.

Phil

**Comment by ewels** [comment]
Addressed in #71
